release notes/v0.42.0.md

```diff
@@ -0,0 +1,152 @@
+k6 v0.42.0 is here and it brings a few exciting updates! :tada:
+
+## Breaking changes
+
+- [#2712](https://github.com/grafana/k6/pull/2712) `k6/ws` returns an HTTP response for failed connections instead of an undefined behavior. Thanks, @brietaylor.
+
+## New Features
+
+### Experimental Prometheus Remote Write Output [#2784](https://github.com/grafana/k6/pull/2784)
+
+This release brings a new builtin Output to any [Prometheus Remote Write](https://docs.google.com/document/d/1LPhVRSFkGNSuU1fBd81ulhsCPR4hkSZyyBj1SZ8fWOM/edit) implementations (e.g. [Prometheus](https://prometheus.io/docs/prometheus/latest/feature_flags/#remote-write-receiver), [Mimir](https://grafana.com/docs/mimir/latest/operators-guide/reference-http-api/#remote-write)). This is an experimental feature, and future releases could introduce breaking changes.
+
+The following example uses `k6 run` with the new output. It uses the defaults options, such as the Remote Write server URL (http://localhost:9090/api/v1/write):
+
+```sh
+k6 run -o experimental-prometheus-rw script.js
+```
+
+It supports the new and convenient experimental [Native Histogram](https://prometheus.io/docs/practices/histograms) feature, added in Prometheus `v2.40.0`. It's not enabled by default, but we expect to make it the default way to map k6 [Trend metrics](https://k6.io/docs/javascript-api/k6-metrics/trend) once the Prometheus' project signals that its mature enough and when more Remote Write implementations support it. For now, if you want to use it, you need to set `K6_PROMETHEUS_RW_TREND_AS_NATIVE_HISTOGRAM` to `true`.
+
+You can find [complete documentation](https://k6.io/docs/results-output/real-time/prometheus-rw) with more examples, use cases, and available configurations.
+
+### More features for the experimental websockets module [#2786](https://github.com/grafana/k6/pull/2786) 
+
+The `k6/experimental/websockets` module that we announced in [v0.40.0](https://github.com/grafana/k6/releases/tag/v0.40.0) release got an update that extends its functionality.
+
+It brings some useful features that `k6/ws` module has, like cookies, custom headers, compression and tags customization support, and the additional syntax of defining the event handlers (`onopen`, `onmessage`, etc) and `ping`/`pong` functionality.
+
+<details>
+<summary> Expand to see an example of the new WebSockets functionality</summary>
+
+This example customizes tags for a WebScoket connection, sets up handlers using the new `on*` syntax, and demonstrates a `ping`/`pong` feature.
+
+```javascript
+import { WebSocket } from "k6/experimental/websockets"
+import { setTimeout, clearTimeout, setInterval, clearInterval } from "k6/experimental/timers"
+
+const CLOSED_STATE = 3
+
+export default function () {
+    var url = "ws://localhost:10000";
+    var params = { "tags": { "my_tag": "hello" } };
+
+    let ws = new WebSocket(url, null, params)
```

Comment:
Hate to point out style issues :sweat_smile:, but this example doesn't use semi-colons and `var`/`let` consistently.

Let's use semi-colons everywhere, `const` where possible, and `let` everywhere else.

Also, maybe 2-space indentation, which is mostly the standard in JS. And, yes, we should update our sample scripts...

Reply:
It should be fixed by https://github.com/grafana/k6/pull/2811/commits/490cd786f1d3ebe201a2e425ea84bb080ff2b727